arduino/health_monitor/health_monitor.ino

```diff
@@ -264,18 +264,18 @@ void setup() {
 
         // Run game update for all enqueued events
         game_update(
-            g_player_state_persistent,
+            globals::player_state_persistent,
             // get next event
             [&event_queue]() -> Event {
```

Comment:
I do wonder why `game_update` isn't operating on the queue directly (aside for the `globals::player_state_persistent` update). I think that would allow https://github.com/kestrel-labs/superspreader/blob/main/arduino/health_monitor/health_monitor_core.h#L156-L163 to become a range-for

Reply:
To poll for changes about a interrupt flag between other event updates...

Otherwise, the interrupt needs to change the queue directly which could lead to fun... Or, it could have lead to more fun when I was using a vector

Reply:
In general, the "give me next or null" even callback is more general/flexible in a function that's already templated for other reasons

Reply:
It could also be an iterator or range, but given the fact that we also have to watch for the interrupt, it would be more complicated for no payoff